src/internal/client.ts

```diff
@@ -1031,4 +1034,91 @@ export class TypedClient {
     const xmlResult = await readAsString(httpRes)
     return xmlParsers.parseReplicationConfig(xmlResult)
   }
+
+  getObjectLegalHold(
+    bucketName: string,
+    objectName: string,
+    getOpts?: GetObjectLegalHoldOptions,
+    callback?: ResultCallback<LegalHoldStatus>,
+  ): Promise<LegalHoldStatus>
+  async getObjectLegalHold(
```

Comment:
Do we need the async overload here?

Reply:
yes， only overload are emitted in type definition

Reply:
Oh, you are correct. 

I didn't notice that callback api signature has been remove, then this async override is not necessary